maven-java/kalix-maven-plugin/src/main/java/kalix/DeployMojo.java

```diff
@@ -36,30 +35,85 @@ public class DeployMojo extends AbstractMojo {
     @Parameter(property = "kalixContext")
     private String kalixContext;
 
+    @Parameter(property = "kalixCurrentProject", required = true)
+    private String kalixCurrentProject;
+
+    @Parameter(defaultValue = "false", property = "kalixDeploymentOn", required = true)
+    private Boolean kalixDeploymentOn;
+
     @Parameter(defaultValue = "30000", property = "cliTimeoutMs", required = true)
     private Long cliTimeoutMs;
 
     private final Log log = getLog();
 
     /**
-     * We deploy by invoking the services deploy command
+     * We deploy by invoking the services deploy command only when `kalixDeploymentOn` is `true`
+     * and the current Kalix project matches with 'kalixCurrentProject'.
      */
     public void execute() throws MojoExecutionException {
-        log.info("Deploying project to Kalix");
+       final List<String> commandLine;
+       final int deploymentResult = 0;
+       if (!Boolean.valueOf(kalixDeploymentOn)){
+           log.info("The variable `kalixDeploymentOn` hasn't been set to `true`. Therefore, not deploying to Kalix");
+           log.info("To deploy to Kalix via `mvn deploy` you need to set `kalixDeploymentOn` to `true` " +
+                   "and `kalixCurrentProject` to your current Kalix project (`kalix config get-project`)");
```

Comment:
```suggestion
                   "and `kalixCurrentProject` to your current Kalix project name (`kalix config get-project`)");
```

Reply:
This is a good point. I was using the ID without much thinking but how about this?
The user creates a project named "foo" then deletes it and then creates a similar project also called "foo". If we compare the name, the user could deploy to a project with the same name but that is not the same project. 
Another way to put it is that the ID is unique but the name is not. How do you find it?

Reply:
I think it should be the name. ID is not so user friendly. People refer to their projects by name. 

We can also see it the other way around. You create two environments. One for staging and one for production. 

CI builds your project and deploys it to `foo` in stage. You promote to production and CI deploys it to `foo` in prod.

Reply:
> We can also see it the other way around. You create two environments. One for staging and one for production.
> 
> CI builds your project and deploys it to `foo` in stage. You promote to production and CI deploys it to `foo` in prod.

Compelling 👍